lib/parallel_tests/cli.rb

```diff
@@ -182,6 +190,23 @@ def parse_options!(argv)
       options
     end
 
+    def parse_additional_test_options(argv)
+      file_args_at = (argv.rindex("--") || -1) + 1
+      files = argv[file_args_at..-1]
+
+      test_opts_at = (argv.index("--") || -2) + 1
+      test_opts_at = 0 if test_opts_at == file_args_at
+
+      test_opts_end = file_args_at - 1
+
+      test_options = argv[test_opts_at...test_opts_end] || []
+
+      [
+        (test_options if test_options.any?),
+        (files if files.any?)
+      ]
+    end
```

Comment:
I considered both slice and partition previously, but they obscured what was happening too much - and made off-by-one errors harder to spot.

Instead, I broke things up into 3 methods (to reduce the clutter in the main method as well).

It's important for readability to separate extracting the options from extracting the files, because parallel_tests needs the second '--' to extract the paths without having to understand RSpec options (or whatever test tool). This means files have to be extracted first, everything else is options.

I wanted the code to reflect that - and searching for the dash was key here (so I named it so).

Currently, it's also a cleaner way to skip the dashes themselves, without resorting to slice or partition or whatever.